internal/handlers/common/params.go

```diff
@@ -57,6 +57,25 @@ func GetOptionalParam[T types.Type](doc *types.Document, key string, defaultValu
 	return res, nil
 }
 
+// GetCompositeParam returns doc's value of composite data type for key or protocol error for missing or invalid parameter.
+func GetCompositeParam(doc *types.Document, key string) (zero any, err error) {
```

Comment:
Let's not use named result parameters as we don't really need them.

Reply:
fixed.